src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -52,6 +52,23 @@ public FileTextLoaderOptionsProvider()
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     public class FileTextLoader : TextLoader
     {
+        /// <summary>
+        /// Guards access to <see cref="_workspace"/> and <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private readonly object _gate = new object();
+
+        /// <summary>
+        /// This is the workspace used to initialize <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private Workspace? _workspace;
```

Comment:
Is there a risk of this becoming a GC root if we're sharing text loaders between a regular workspace and some more ephemeral workspace like a PreviewWorkspace?

Reply:
It's a possibility, but the code to avoid that is substantially more complex.

Reply:
Yeah, the loaders will get shared to any forked Workspace -- they're held by the Solution snapshot.